src/System.Net.Sockets/src/System/Net/Sockets/BaseOverlappedAsyncResult.Windows.cs

```diff
@@ -139,18 +149,13 @@ private void CompletionCallback(int numBytes, SocketError socketError)
             InvokeCallback(result);
         }
 
-        // The following property returns the Win32 unsafe pointer to
-        // whichever Overlapped structure we're using for IO.
-        internal SafeNativeOverlapped OverlappedHandle
-        {
-            get
-            {
-                // On WinNT we need to use (due to the current implementation)
-                // an Overlapped object in order to bind the socket to the
-                // ThreadPool's completion port, so return the native handle
-                return _nativeOverlapped == null ? SafeNativeOverlapped.Zero : _nativeOverlapped;
-            }
-        }
+        // The following property returns a SafeHandle for the pointer to
+        // the Overlapped structure we're using for IO.
+        internal SafeNativeOverlapped OverlappedHandle =>
```

Comment:
I think you should be able to remove this -- all uses of this should be replaced with the new DangerousOverlappedPointer below.